### PR TITLE
[Merged by Bors] - feat(Data/Set): monotonicity on insert

### PR DESCRIPTION
**Pull request by b-mehta**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Function.lean [hunk 1]
```diff
@@ -1551,6 +1551,16 @@ theorem StrictMono.codRestrict [Preorder α] [Preorder β] {f : α → β} (hf :
     {s : Set β} (hs : ∀ x, f x ∈ s) : StrictMono (Set.codRestrict f s hs) :=
   hf
 
+lemma strictMonoOn_insert_iff [Preorder α] [Preorder β] {f : α → β} (s : Set α) (a : α) :
```
Comment (urkud) [inline Mathlib/Data/Set/Function.lean:1554]:
```suggestion
lemma strictMonoOn_insert_iff [Preorder α] [Preorder β] {f : α → β} {s : Set α} {a : α} :
```
same below.

**Comment by urkud** [comment]
Please add lemmas about `MonotoneOn`/AntitoneOn` as well.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Comment by sgouezel** [comment]
bors r-
Can you merge master, fix the build, and send it back to bors?
bors d+

**Pull request closed without merging**